### PR TITLE
add pycon support for markdown

### PR DESCRIPTION
**Pull request by Spectre5**
Implements #74

Addes new regex `MD_PYCON_RE` and refactors  `_rst_pycon_match` to `_pycon_match` for use in both `_md_pycon_match` and `_rst_pycon_match`.

~I also added Python 3.8 to the tox test environment.~

### CODE REVIEW

**Comment by Spectre5** [comment]
Note the test `test_format_src_markdown_pycon()` which shows that this will remove the whitespace before the prompt or continuation line to make the prompt line up with the first ` before ```pycon.  I believe that it should align, at least in GitHub's flavor it needs to for correct syntax highlighting.

File: tests/blacken_docs_test.py [hunk 1]
```diff
@@ -725,3 +725,243 @@ def test_format_src_rst_pycon_comment_before_promopt():
...
+
+def test_format_src_markdown_pycon_comment_before_promopt():
+    before = (
+        '```pycon\n'
+        '# Comment about next line\n'
+        '>>> pass\n'
+        '```\n'
+    )
+    after, _ = blacken_docs.format_str(before, BLACK_MODE)
+    assert after == before
```
Comment (asottile) [inline tests/blacken_docs_test.py:967]: you don't need all these tests, please just one test demonstrating your change

Comment (Spectre5) [reply tests/blacken_docs_test.py]: I've kept just the first 1

**Review by asottile — APPROVED** [review]

**Pull request merged**